runtime-interface/src/lib.rs

```diff
@@ -96,36 +104,57 @@ pub trait GearRI {
         gas_left: (GasLeft,),
     ) -> (GasLeft, Result<(), ProcessAccessError>) {
         let mut gas_left = gas_left.0;
-        let res = lazy_pages::pre_process_memory_accesses(reads, writes, &mut gas_left);
+        let gas_before = gas_left.gas;
+        let res = lazy_pages::pre_process_memory_accesses(reads, writes, &mut gas_left.gas);
+
+        // Support charge for allowance otherwise DB will be corrupted.
+        gas_left.allowance = gas_left
+            .allowance
+            .saturating_sub(gas_before.saturating_sub(gas_left.gas));
+
         (gas_left, res)
     }
 
+    #[version(2)]
+    fn pre_process_memory_accesses(reads: &[u8], writes: &[u8], gas_bytes: &mut [u8; 8]) -> u8 {
```

Comment:
```suggestion
    fn pre_process_memory_accesses(reads: &[u8], writes: &[u8], gas_bytes: &mut u64) -> u8 {
```
Isn't primitive?

Reply:
Only u8 is supported

Reply:
https://paritytech.github.io/substrate/master/sp_runtime_interface/host/trait.IntoPreallocatedFFIValue.html#foreign-impls